client/src/components/Profile/Profile.js

```diff
@@ -1,8 +1,39 @@
-import React from "react";
+import { React, useState } from "react";
 import PropTypes from "prop-types";
+import * as fetch from "../../lib/fetch";
 
 function Profile({ id }) {
-  return <>Profile of user with id {id}</>;
+  const [userExists, setUserExists] = useState(true);
+  const [name, setName] = useState("");
+  const [userID, setUserID] = useState("");
+
+  const getData = async () => {
+    const { ok, resJSON } = await fetch.getUserData(id);
+    return { ok, resJSON };
+  };
+
+  getData().then(({ ok, resJSON }) => {
```

Comment:
Can we put the fetching into a `useEffect` hook?

```js
useEffect(() => {
  (async () => {
    const { resJSON, ok } = await fetcher.getUserData(id);
    if (!ok) {
      setUserExists(false);
      return;
    }
    setName(resJSON.name);
  })();
}, [id]);
```

Reply:
Good idea, I forgot that exists :')